doc/rbd/rbd-mirroring.rst

```diff
@@ -0,0 +1,233 @@
+
+For example::
+
+        rbd --cluster local mirror image enable image-pool/image-1
+
+Disable Image Mirroring
+-----------------------
+
+To disable mirroring for a specific image with ``rbd``, specify the
+``mirror image disable`` command along with the pool and image name::
+
+        rbd mirror image disable {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster local mirror image disable image-pool/image-1
+
+Image Promotion and Demotion
+----------------------------
+
+An image can only be primary within a single Ceph cluster.  Attempting to
+access a non-primary image will result in an error.  In a failover scenario,
+access to the primary image should be stopped (e.g. power down the VM or remove
+the associated drive from a VM), demote the current primary image, promote
+the new primary image, and resume access to the image on the alternate
+cluster.
+
+To demote an image to secondary with ``rbd``, specify the ``mirror image demote``
+command along with the pool and image name::
+
+        rbd mirror image demote {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster local mirror image demote image-pool/image-1
+
+To promote an image to primary with ``rbd``, specify the ``mirror image promote``
+command along with the pool and image name::
+
+        rbd mirror image promote {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster remote mirror image promote image-pool/image-1
+
+.. tip:: Since the primary / secondary status is per-image, it is possible to
+   have two clusters split the IO load and stage failover / failback.
+.. note:: Promotion can be forced using the ``--force`` option.  This will result
+   in a split-brain scenario between the two peers and the image will no longer
+   be in-sync until a `force resync command`_ is issued.
+
+Force Image Resync
+------------------
+
+If a split-brain event is detected by the ``rbd-daemon``, it will not attempt
+to mirror the affected image until corrected.  To resume mirroring for an image,
+first `demote the image`_ determined to be out-of-date and then request a resync
+to the primary image.  To request an image resync with ``rbd``, specify the
+``mirror image resync`` command along with the pool and image name::
+
+        rbd mirror image resync {pool-name}/{image-name}
+
+For example::
+
+        rbd mirror image resync image-pool/image-1
+
+.. note:: The ``rbd`` command only flags the image as requiring a resync. The
+   local cluster's ``rbd-mirror`` daemon process is responsible for performing
+   the resync asynchronously.
+
+rbd-mirror Daemon
+=================
+
+The ``rbd-mirror`` daemon is responsible for watching image journals on a
+remote, peer cluster and replaying the journal events against the local
+cluster.  The journal events recorded by the RBD image journaling feature
+include all modifications to the image in-order to ensure that a
+crash-consistent mirror of the remote image is available locally
```

Comment:
s/locally/locally./ but I suggest rewording the last sentence like so:

The RBD image journaling feature records all modifications to the image in the order they occur. This ensures that a crash-consistent mirror of the remote image is available locally.